scrape/metadata_watcher.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scrape
```

Comment:
What's the thought behind putting this in the scrape package? I don't see it using anything that is not exposed.

Reply:
I tried to make it similar to `wal.Watcher` and put it close to what it observes. It's true that the types returned are exposed, but felt strange to create a struct within a package which only purpose was to return types from another package.

That's what my gut feeling told me and I certainly don't have a strong opinion on where it goes. Everything is one file, in the event of change/removal you would still only have to modify one file.

Reply:
I'm with Chris, I think this belongs in the remote package; I doubt we'll see this reused anywhere else.

Reply:
done!